flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/failover/ExponentialDelayRestartBackoffTimeStrategyTest.java

```diff
@@ -73,14 +75,14 @@ void testMaxBackoff() throws Exception {
     }
 
     @Test
-    void testResetBackoff() throws Exception {
+    void testResetBackoff() {
         final long initialBackoffMS = 1L;
         final long resetBackoffThresholdMS = 8L;
         final ManualClock clock = new ManualClock();
 
         final ExponentialDelayRestartBackoffTimeStrategy restartStrategy =
                 new ExponentialDelayRestartBackoffTimeStrategy(
-                        clock, initialBackoffMS, 5L, 2.0, resetBackoffThresholdMS, 0.25);
+                        clock, initialBackoffMS, 5L, 2.0, resetBackoffThresholdMS, 0.25, 100);
```

Comment:
Could you help explain why set the `attemptsBeforeResetBackoff` to 100 here?

Reply:
It doesn't have strong reason, I just set it to a big value to ensure this test doesn't reach this limitation.

Reply:
Update `attemptsBeforeResetBackoff` to `Integer.MAX_VALUE` for all tests not related to `maxAttempts`.

`Integer.MAX_VALUE` makes it easier for other developers to know that attemptsBeforeResetBackoff does not take effect.